RealmSwift/SectionedResults.swift

```diff
@@ -0,0 +1,1050 @@
+     Any insertion or removal to the `Dog` type collection being observed
+     would also trigger a notification.
+     - If the above example observed the `["toys"]` key path, then any insertion,
+     deletion, or modification to the `toys` list for any element in the collection
+     would trigger the block.
+     Changes to any value on any `Toy` that is linked to a `Dog` in this collection
+     would *not* trigger the block.
+     Any insertion or removal to the `Dog` type collection being observed
+     would still trigger a notification.
+
+     - note: Multiple notification tokens on the same object which filter for
+     separate key paths *do not* filter exclusively. If one key path
+     change is satisfied for one notification token, then all notification
+     token blocks for that object will execute.
+
+     You must retain the returned token for as long as you want updates to be sent to the block. To stop receiving
+     updates, call `invalidate()` on the token.
+
+     - warning: This method cannot be called during a write transaction, or when the containing Realm is read-only.
+
+     - parameter keyPaths: Only properties contained in the key paths array will trigger
+                           the block when they are modified. If `nil`, notifications
+                           will be delivered for any property change on the object.
+     - parameter queue: The serial dispatch queue to receive notification on. If
+                        `nil`, notifications are delivered to the current thread.
+     - parameter block: The block to be called whenever a change occurs.
+     - returns: A token which must be held for as long as you want updates to be delivered.
+     */
+    public func observe<T: ObjectBase>(keyPaths: [PartialKeyPath<T>],
+                                       on queue: DispatchQueue? = nil,
+                                       _ block: @escaping (RealmSectionedResultsChange<Self>) -> Void) -> NotificationToken {
+        return collection.addNotificationBlock(wrapObserveBlock(block), keyPaths: keyPaths.map(_name(for:)), queue: queue)
+    }
+
+    /// :nodoc:
+    public func makeIterator() -> RLMSectionedResultsIterator<Key, T> {
+        return RLMSectionedResultsIterator(collection: collection)
+    }
+
+    /// :nodoc:
+    public static func == (lhs: SectionedResults<Key, T>, rhs: SectionedResults<Key, T>) -> Bool {
+        return lhs.collection == rhs.collection
+    }
+}
+
+
+/// `ResultsSection` is a collection which allows access  to objects that belong to a given section key.
+/// The collection is lazily evaluated, meaning that if the underlying collection has changed a full recalculation of the section keys will take place.
+/// A `ResultsSection` instance can be observed and it also conforms to `ThreadConfined`.
+public struct ResultsSection<Key: _Persistable & Hashable, T: RealmCollectionValue>: RealmSectionedResultImpl, Identifiable {
```

Comment:
`Identifiable` has a deployment target of iOS 13 and so needs to be in an extension guarded by `@available`. I think the reason it doesn't give any warnings here is that our tests are built with a higher deployment target.